libcloud/compute/drivers/dimensiondata.py

```diff
@@ -0,0 +1,594 @@
+    def __init__(self, code, msg, driver):
+        self.code = code
+        self.msg = msg
+        self.driver = driver
+
+    def __str__(self):
+        return "%s: %s" % (self.code, self.msg)
+
+    def __repr__(self):
+        return ("<DimensionDataAPIException: code='%s', msg='%s'>" %
+                (self.code, self.msg))
+
+
+class DimensionDataConnection(ConnectionUserAndKey):
+    """
+    Connection class for the DimensionData driver
+    """
+
+    host = 'api-na.dimensiondata.com'
+    api_path = '/oec'
+    api_version = '0.9'
+    _orgId = None
+    responseCls = DimensionDataResponse
+
+    allow_insecure = False
+
+    def add_default_headers(self, headers):
+        headers['Authorization'] = \
+            ('Basic %s' % b64encode(b('%s:%s' % (self.user_id,
+                                                 self.key))).decode('utf-8'))
+        return headers
+
+    def request(self, action, params=None, data='',
+                headers=None, method='GET'):
+        action = "%s/%s/%s" % (self.api_path, self.api_version, action)
+
+        return super(DimensionDataConnection, self).request(
+            action=action,
+            params=params, data=data,
+            method=method, headers=headers)
+
+    def request_with_orgId(self, action, params=None, data='',
+                           headers=None, method='GET'):
+        action = "%s/%s" % (self.get_resource_path(), action)
+
+        return super(DimensionDataConnection, self).request(
+            action=action,
+            params=params, data=data,
+            method=method, headers=headers)
+
+    def get_resource_path(self):
+        """
+        This method returns a resource path which is necessary for referencing
+        resources that require a full path instead of just an ID, such as
+        networks, and customer snapshots.
+        """
+        return ("%s/%s/%s" % (self.api_path, self.api_version,
+                              self._get_orgId()))
+
+    def _get_orgId(self):
+        """
+        Send the /myaccount API request to DimensionData cloud and parse the
+        'orgId' from the XML response object. We need the orgId to use most
+        of the other API functions
+        """
+        if self._orgId is None:
+            body = self.request('myaccount').object
+            self._orgId = findtext(body, 'orgId', DIRECTORY_NS)
+        return self._orgId
+
+
+class DimensionDataStatus(object):
+    """
+    DimensionData API pending operation status class
+        action, requestTime, username, numberOfSteps, updateTime,
+        step.name, step.number, step.percentComplete, failureReason,
+    """
+    def __init__(self, action=None, requestTime=None, userName=None,
+                 numberOfSteps=None, updateTime=None, step_name=None,
+                 step_number=None, step_percentComplete=None,
+                 failureReason=None):
+        self.action = action
+        self.requestTime = requestTime
```

Comment:
Heh, looks like this stylistic thing slipped through when we originally reviewed and merged opsource driver.

Variables should use underscore separated names and not camel case. It seems that now that we are adding a new driver is a good time to fix that.